packages/shared/lib/src/sdk/mod.rs

```diff
@@ -90,9 +97,20 @@ impl Sdk {
     ) -> Result<(), JsError> {
         let args = tx::bond_tx_args(tx_msg, password)?;
 
-        namada::ledger::tx::submit_bond(&mut self.client, &mut self.wallet, args)
+        let (mut tx, _, pk) =
+            namada::ledger::tx::build_bond(&mut self.client, &mut self.wallet, args.clone())
+                .await
+                .map_err(|e| JsError::from(e))?;
```

Comment:
I've learned recently that you can pass function as a rust closure so:
`.map_err(JsError::from)?;`
should be enough

Reply:
Nice! Somehow I forgot about that feature

Reply:
Done!